includes/admin/feedzy-rss-feeds-log.php

```diff
@@ -0,0 +1,858 @@
+					$t = fgetc( $handle );
+					if ( false === $t ) {
+						$beginning = true;
+						break;
+					}
+					--$pos;
+				}
+				--$line_counter;
+				if ( $beginning ) {
+					rewind( $handle );
+				}
+				$line = fgets( $handle );
+				if ( false !== $line ) {
+					$text[] = $line;
+				}
+				if ( $beginning ) {
+					break;
+				}
+			}
+			fclose( $handle );
+		
+			return $text;
+		} catch ( Throwable $e ) {
+			return new WP_Error( 'error_reading_log_file', $e->getMessage() );
+		}
+	}
+
+	/**
+	 * Read the last N lines using WP_Filesystem (compatible with all filesystem types).
+	 *
+	 * @since 5.1.0
+	 * @param int $lines_count The number of lines to read.
+	 * @return array<string>|WP_Error The last N lines.
+	 */
+	private function read_last_lines_wp_filesystem( $lines_count ) {
+		if ( ! $this->log_file_exists() || ! $this->is_log_readable() ) {
+			return new WP_Error( 'log_file_not_found' );
+		}
+
+		$content = $this->filesystem->get_contents( $this->filepath );
+		if ( false === $content ) {
+			return new WP_Error( 'unable_to_read_log_file' );
+		}
+
+		$lines = explode( "\n", $content );
+		$lines = array_filter(
+			$lines,
+			function ( $line ) {
+				return '' !== trim( $line );
+			}
+		);
+
+		return array_reverse( array_slice( $lines, -$lines_count ) );
+	}
+
+	/**
+	 * Delete the log file if it is older than 14 days or if the size exceeds the maximum allowed size.
+	 *
+	 * @since 5.1.0
+	 * @return bool
+	 */
+	public function should_clean_logs() {
+		if ( ! file_exists( $this->filepath ) ) {
+			return false;
+		}
+
+		$is_too_big = $this->get_log_file_size() > self::DEFAULT_MAX_FILE_SIZE;
+		$is_too_old = filemtime( $this->filepath ) < strtotime( '-14 days' );
+
+		if ( $is_too_big || $is_too_old ) {
+			return $this->delete_log_file();
+		}
+
+		return false;
+	}
+
+	/**
+	 * Static convenience methods.
+	 *
+	 * @since 5.1.0
+	 * @param string            $name The method name.
+	 * @param array<int, mixed> $arguments The method arguments.
+	 * @return mixed
+	 */
+	public static function __callStatic( $name, $arguments ) {
+		$instance = self::get_instance();
+		if ( method_exists( $instance, $name ) ) {
+			return call_user_func_array( array( $instance, $name ), $arguments );
+		}
+	}
+
+	/**
+	 * Register REST API endpoints for logs.
+	 *
+	 * @since 5.1.0
+	 * @return void
+	 */
+	public function register_endpoints() {
+		register_rest_route(
+			'feedzy/v1',
+			'/logs/download',
+			array(
+				'methods'             => 'GET',
+				'callback'            => array( $this, 'export_logs_endpoint' ),
+				'permission_callback' => function () {
+					return current_user_can( 'manage_options' );
+				},
+			)
+		);
+		register_rest_route(
+			'feedzy/v1',
+			'/logs/delete',
+			array(
+				'methods'             => 'GET',
+				'callback'            => array( $this, 'delete_log_file_endpoint' ),
+				'permission_callback' => function () {
+					return current_user_can( 'manage_options' );
+				},
+			)
+		);
+	}
```

Comment:
I would make the second route either a POST or a DELETE request as it is not a read operation.